source_control/gitlab_project.py

```diff
@@ -0,0 +1,389 @@
+#!/usr/bin/python
+# (c) 2015, Werner Dijkerman (ikben@werner-dijkerman.nl)
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: gitlab_project
+short_description: Creates/updates/deletes Gitlab Projects
+description:
+   - When the project does not exists in Gitlab, it will be created.
+   - When the project does exists and state=absent, the project will be deleted.
+   - When changes are made to the project, the project will be updated.
+version_added: "2.0"
+author: Werner Dijkerman (@dj-wasabi)
+requirements:
+    - pyapi-gitlab python module
+options:
+    server_url:
+        description:
+            - Url of Gitlab server, with protocol (http or https).
+        required: true
+        default: null
```

Comment:
`default: null` can be removed if `required: true`. There is no default, if it requires input.